src/librustc_middle/query/mod.rs

```diff
@@ -482,7 +527,9 @@ rustc_queries! {
     }
 
     TypeChecking {
-        query has_typeck_tables(_: DefId) -> bool {}
+        query has_typeck_tables(def_id: DefId) -> bool {
+            desc { |tcx| "checking whether `{}` has a body", tcx.def_path_str(def_id) }
```

Comment:
Is it limited to bodies? If so, it should probably be renamed (open an issue I guess). Otherwise, it should say "can be type-checked" or "has type-checking results".

Oh this reminds me that I've wanted `TypeckTables` to become `TypeckResults` for a while now.

Reply:
`has_typeck_tables` is essentially `primary_body_of(tcx, id).is_some()`, so, yes, this is limited to crate-local items which have bodies.